.semaphore/semaphore.yml

```diff
@@ -155,7 +155,7 @@ after_pipeline:
       - name: SonarQube
         commands:
           - checkout
-          - sem-version java 11
+          - if [[ $SEMAPHORE_GIT_BRANCH =~ ^7\..* ]]; then sem-version java 8; else sem-version java 17; fi
```

Comment:
why change from 11 to 8? 11 is the recommended version for CP.
should we cover 6.x releases as well in case of backports?

Reply:
For 7.x, we still support Java 8. -- I don't think it's about recommended, but supported?

It's more a question for @hk10111 thouhg. Cf https://github.com/confluentinc/kafka-streams-examples/pull/527

I would have hard-coded Java 8 for older branches and hard-coded Java 17 for 8.0+, but they want to have the same `semaphore.yml` config across all branches -- that's why the if-else.

Guess we could also use Java 11 for older branches, but historically we use Java 8.

Anyway, the other commit finally made it into all branches I believe, so this change will go away for this PR anyway.